temporal-sdk/src/main/java/io/temporal/internal/worker/LocalActivityWorker.java

```diff
@@ -260,8 +262,56 @@ public boolean dispatch(
           new LocalActivityExecutionContext(
               params, resultCallback, localRetryDeadline, scheduleToCloseDeadline);
 
-      LocalActivityAttemptTask task =
-          new LocalActivityAttemptTask(executionContext, params.getInitialActivityTask());
+      PollActivityTaskQueueResponse activityTask = params.getInitialActivityTask();
+      LocalActivityAttemptTask task = new LocalActivityAttemptTask(executionContext, activityTask);
+
+      final Failure previousExecutionFailure = executionContext.getPreviousExecutionFailure();
+      if (previousExecutionFailure != null) {
```

Comment:
Maybe consider breaking the inner part of the if block out into a function

Reply:
Yeah, I will work on readability & doc side of this PR more.